Library/Homebrew/utils/versions.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require "open3"
+require "formula"
+
+module Versions
+  def current_formula_version(formula_name)
+    Formula[formula_name].version.to_s.to_f
+  end
+
+  def bump_formula_pr(formula_name, url)
+    command_args = [
+      "brew",
+      "bump-formula-pr",
+      "--no-browse",
+      "--dry-run",
+      formula_name,
+      "--url=#{url}",
+    ]
+
+    response = Open3.capture2e(*command_args)
+    parse_formula_bump_response(response)
+  end
+
+  def parse_formula_bump_response(formula_bump_response)
+    response, _status = formula_bump_response
+    response
+  end
+
+  def check_for_open_pr(formula_name, download_url)
+    ohai "- Checking for open PRs for formula : #{formula_name}"
+
+    response = bump_formula_pr(formula_name, download_url)
+    !response.include? "Error: These open pull requests may be duplicates"
```

Comment:
Would be nice to do refactor this check into a separate function and call it rather than shelling out to `bump-formula-pr` to get this.